core/api/src/services/openai/index.ts

```diff
@@ -0,0 +1,114 @@
+import OpenAI, { OpenAIError } from "openai"
+
+import { env } from "@/config/env"
+import { sleep } from "@/utils"
+import { UnknownDomainError } from "@/domain/shared"
+
+let openai: OpenAI
+
+if (env.OPENAI_API_KEY) {
+  openai = new OpenAI({
+    apiKey: env.OPENAI_API_KEY,
+  })
+}
+
+export const Assistant = () => {
+  const assistantId = env.OPENAI_ASSISTANT_ID
+  if (!assistantId) {
+    throw new Error("No assistant id found")
+  }
+  if (!openai) {
+    throw new Error("No openai instance found")
+  }
+
+  const getMessages = async (threadId: ThreadId): Promise<Message[] | Error> => {
+    try {
+      const messages = await openai.beta.threads.messages.list(threadId)
+
+      const processedResponse = messages.data
+        .map((item) => ({
+          id: item.id,
+          role: item.role,
+          message:
+            item.content[0]?.type !== "text"
+              ? "message is not a text" // TODO: proper error handling
+              : item.content[0].text.value,
+          timestamp: item.created_at,
+        }))
+        .reverse()
+
+      return processedResponse
+    } catch (err) {
+      if (err instanceof OpenAIError) {
+        return err
+      }
```

Comment:
why this? if this is from openai library we need to create a proper wrapper (domain error)